src/tools/annotation/bidirectionalTool/addNewMeasurement.js

```diff
@@ -22,15 +23,32 @@ export default function(evt, interactionType) {
     measurementData.active = false;
     external.cornerstone.updateImage(element);
 
-    // Trigger measurement modified event
-    const eventType = EVENTS.MEASUREMENT_MODIFIED;
-    const modifiedEventData = {
-      toolName: this.name,
-      element,
-      measurementData,
+    const measurementModifiedHandler = () => {
+      const modifiedEventData = {
+        toolName: this.name,
+        element,
+        measurementData,
+      };
+
+      calculateLongestAndShortestDiameters(eventData, measurementData);
+
+      external.cornerstone.triggerEvent(
+        element,
+        EVENTS.MEASUREMENT_MODIFIED,
+        modifiedEventData
+      );
+
+      element.removeEventListener(
+        external.cornerstone.EVENTS.IMAGE_RENDERED,
+        measurementModifiedHandler
+      );
     };
 
-    external.cornerstone.triggerEvent(element, eventType, modifiedEventData);
+    // Wait on image render before triggering the modified event
+    element.addEventListener(
+      external.cornerstone.EVENTS.IMAGE_RENDERED,
+      measurementModifiedHandler
```

Comment:
FWIW, this tool method is called for every _IMAGE_RENDERED_ event already:

https://github.com/cornerstonejs/cornerstoneTools/blob/023be5f38788a071384bcc677019880f370c6018/src/tools/annotation/bidirectionalTool/renderToolData.js#L18

Reply:
Just removed the use of it.